active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,262 @@
+---
+REP Number: 0002
+Author: asilverman (Ariel Silverman)
+Start Date: 2023-12-19
+Feature Status: Private Preview
+Bicep Issue Number(s): [12202](https://github.com/Azure/bicep/issues/12202)
+---
+
+# Title - Centralized management of resource type provider versions using package.json
+
+## Summary
+
+This document proposes a feature that enables users to manage Bicep resource type provider versions from a central location. This enhances convenience by sparing the user from making changes in multiple files in order to upgrade a dynamically loaded resource types provider package version.
+
+## Terms and definitions
+
+- **Resource Types Provider Package aka provider package** - The OCI Registry artifact that contains the resource type definitions used by Bicep to identify the resource types properties, functions and other relevant type information that it uses to generate ARM-JSON.
+- **types.json protocol** - It is a serialization protocol used by Bicep to encode type information as data that can be loaded to the Bicep runtime.
+
+## Motivation
+
+The [resource type providers dynamic loading feature](https://github.com/Azure/bicep/issues/5453) introduces a change in provider declaration statement syntax to include the registry address and version that Bicep will use to resolve resource type declarations.
+
+**Provider Declaration Syntax**
+
+```bicep
+provider 'br:mcr.microsoft.com/bicep/providers/az@1.2.3' // fully-qualified
+provider 'br/public:az@1.2.3' // uses provider aliasing defined in bicepconfig.json
+```
+
+We also currently support the following syntax for "built-in" providers:
+
+```bicep
+provider 'sys@1.0.0'
```

Comment:
I would like to point out that `sys` is more like a compile-time namespace import (the feature Jonny has been working on), not a provider, since all it contains are some built-in functions. We used to share the keyword `import` between providers and imports, but now that we changed the keyword for providers, at some point we will need to update the implementation for `sys` to make it a compile-time import. This is out of scope of this REP though.

Reply:
Conceptually I think there's overlap with both imports & providers (at some point we will add function execution to providers, and "az" has functions). This would be an interesting discussion topic, but agree we can consider it out-of-scope for now.

Reply:
Added a call out in the out of scope section regarding this topic.